sanic/response.py

```diff
@@ -97,40 +97,39 @@ def __init__(self, body=None, status=200, headers=None,
     def output(self, version="1.1", keep_alive=False, keep_alive_timeout=None):
         # This is all returned in a kind-of funky way
         # We tried to make this as fast as possible in pure python
-        timeout_header = b''
         if keep_alive and keep_alive_timeout:
-            timeout_header = b'Keep-Alive: timeout=%d\r\n' % keep_alive_timeout
-
+            if 'Keep-Alive' not in self.headers:
+                self.headers['Keep-Alive'] = keep_alive_timeout
+        if 'Connection' not in self.headers:
+            if keep_alive:
+                self.headers['Connection'] = 'keep-alive'
+            else:
+                self.headers['Connection'] = 'close'
+        if 'Content-Length' not in self.headers:
```

Comment:
I could see how this could affect performance, so I changed it to use a default_header dictionary and ChainMap.